pkg/scheduler/flavorassigner/flavorassigner_test.go

```diff
@@ -59,13 +63,16 @@ func TestAssignFlavors(t *testing.T) {
 	}
 
 	cases := map[string]struct {
-		wlPods             []kueue.PodSet
-		wlReclaimablePods  []kueue.ReclaimablePod
-		clusterQueue       cache.ClusterQueue
-		wantRepMode        FlavorAssignmentMode
-		wantAssignment     Assignment
-		enableLendingLimit bool
-		enableFairSharing  bool
+		wlPods                      []kueue.PodSet
+		wlReclaimablePods           []kueue.ReclaimablePod
+		clusterQueue                kueue.ClusterQueue
+		clusterQueueUsage           resources.FlavorResourceQuantitiesFlat
+		clusterQueueGuaranteedQuota resources.FlavorResourceQuantitiesFlat
```

Comment:
this should not be necessary. It should be calculated by the cache logic based on the lendingLimit

Reply:
after removal, a bunch of tests fail. we can clean up in https://github.com/kubernetes-sigs/kueue/issues/2502

Reply:
Maybe some tests didn't put the right LendingLimit.
Ok.

Reply:
I was able to remove guaranteedQuota; I'm not sure what was happening last week (unless I thought you were referring to a different field?)